### PR TITLE
SingleTopNav: Fix losing focus when moving between menu states

### LINKED ISSUE #94593: Unified AppChrome bar: Fix losing keyboard focus between open/closed/docked states
- [x] opening undocked menu
  - expected: focus moves to grafana logo inside the overlay menu
  - actual: focus stays on grafana logo on the main page (can tell because the `open menu` tooltip is still showing)
- [x] closing undocked menu
  - expected: focus moves to grafana logo on the main page
  - actual: focus moves to grafana logo on the main page
- [x] opening docked menu
  - expected: focus moves to grafana logo in the docked menu
  - actual: focus reverts to `body`
- [x] closing docked menu
  - expected: focus moves to grafana logo on the main page
  - actual: focus reverts to `body`
- [x] docking menu
  - expected: focus stays on the `undock menu` button
  - actual: focus stays on the `undock menu` button
- [x] undocking menu
  - expected: focus moves to grafana logo on the main page
    - this is the only one i'm not 100% about. i _think_ this behaviour makes the most sense 👍 
  - actual: focus reverts to `body`

for the overlay behaviour, `restoreFocus` _should_ handle this but it doesn't seem to be quite working correctly 🤔 

p.s. might be helpful to stick a watch expression in your console of `document.activeElement` to show what element is currently focused

### PR DESCRIPTION
**Pull request by ashharrison90**
**What is this feature?**

- correctly moves focus between the various megamenu states (open, closed, docked)

**Why do we need this feature?**

- better a11y for keyboard users

**Who is this feature for?**

- everyone!

**Which issue(s) does this PR fix?**:

Fixes https://github.com/grafana/grafana/issues/94593

**Special notes for your reviewer:**

Please check that:
- [ ] It works as expected from a user's perspective.
- [ ] If this is a pre-GA feature, it is behind a feature toggle.
- [ ] The docs are updated, and if this is a [notable improvement](https://grafana.com/docs/writers-toolkit/contribute/release-notes/#how-to-determine-if-content-belongs-in-whats-new), it's added to our [What's New](https://grafana.com/docs/writers-toolkit/contribute/release-notes/) doc.

### CODE REVIEW

File: public/app/core/components/AppChrome/MegaMenu/MegaMenu.tsx [hunk 1]
```diff
@@ -76,9 +76,11 @@ export const MegaMenu = memo(
       }
 
       // refocus on undock/menu open button when changing state
-      setTimeout(() => {
-        document.getElementById(state.megaMenuDocked ? 'mega-menu-toggle' : 'dock-menu-button')?.focus();
-      });
+      if (!config.featureToggles.singleTopNav) {
+        setTimeout(() => {
+          document.getElementById(state.megaMenuDocked ? 'mega-menu-toggle' : 'dock-menu-button')?.focus();
```
Comment (L-M-K-B) [inline public/app/core/components/AppChrome/MegaMenu/MegaMenu.tsx:81]: Is there a specific reason why you are not using MEGA_MENU_TOGGLE_ID and DOCK_MENU_BUTTON_ID here? If not, please add both.

**Review by L-M-K-B — APPROVED** [review]
I checked the behaviour of the focus locally and it works as expected. There's a tiny nit that I commented on but from my point of view it's not a blocker.

File: public/app/core/components/AppChrome/MegaMenu/MegaMenu.tsx (hunk 1 above)
Comment (ashharrison90) [reply public/app/core/components/AppChrome/MegaMenu/MegaMenu.tsx:81]: only that it's gonna be removed soon anyway 😄 i'll change it...

**Pull request merged**